openscenario/openscenario_interpreter/include/openscenario_interpreter/syntax/condition.hpp

```diff
@@ -49,9 +57,37 @@ struct Condition : public ComplexType
 
   bool current_value;
 
+private:
+  struct History
+  {
+    double time;
+    bool result;
+  };
+
+  std::list<History> histories;
+
+public:
   explicit Condition(const pugi::xml_node & node, Scope & scope);
 
   auto evaluate() -> Object;
+
+private:
+  template <typename... Booleans>
+  auto update_condition(std::function<bool(Booleans...)> condition) -> Object
+  {
+    histories.push_back({evaluateSimulationTime(), ComplexType::evaluate().as<Boolean>()});
+    if (auto iterator = std::find_if(
+          std::begin(histories), std::end(histories),
+          [this](const auto & entry) { return entry.time > histories.back().time - delay; });
+        static_cast<std::ptrdiff_t>(sizeof...(Booleans)) <=
+        std::distance(std::begin(histories), iterator)) {
+      current_value = std::apply(condition, std::tuple{Booleans((--iterator)->result)...});
```

Comment:
This line depends on the order of evaluation of list initialization[^1] and cannot be replaced with direct initialization like `tuple(...)` whose order is *indeterminately sequenced*[^2]. This may be a confusing difference so it would be better to fold over the comma operator[^3].

[^1]: > 10 ) In [list-initialization](https://en.cppreference.com/w/cpp/language/list_initialization), every value computation and side effect of a given initializer clause is sequenced before every value computation and side effect associated with any initializer clause that follows it in the brace-enclosed comma-separated list of initializers.

[^2]: > 15 ) In a function call, value computations and side effects of the initialization of every parameter are indeterminately sequenced with respect to value computations and side effects of any other parameter. 
      > 17 ) Every expression in a comma-separated list of expressions in a parenthesized initializer is evaluated as if for a function call (indeterminately-sequenced)

[^3]: > 9 ) Every value computation and side effect of the first (left) argument of the built-in [comma operator](https://en.cppreference.com/w/cpp/language/operator_other#Built-in_comma_operator) , is sequenced before every value computation and side effect of the second (right) argument.
    
     https://en.cppreference.com/w/cpp/language/eval_order

```suggestion
      auto results = std::tuple<Booleans...>();
      std::apply([&](auto &... results) { ((results = (--iterator)->result), ...); }, results);
      current_value = std::apply(condition, results);
```

Reply:
If the standard specifies the order of argument evaluation for list initialization, then there is nothing to worry about using list initialization.
Your suggestion of using the comma operator isn't bad, but it's still tricky either way.
That's why I prefer the original code you wrote.
In other words, the code below is OK.

```
      current_value = std::apply(condition, std::tuple{Booleans((--iterator)->result)...});
```